### PR TITLE
Gradle properties deprecation

### PR DESCRIPTION
**Pull request by PetrakovichVictoria**

### CODE REVIEW

File: docs/topics/compatibility-guide-16.md [hunk 1]
```diff
@@ -452,3 +452,48 @@ perspective
...
+
+### Deprecate kapt.use.worker.api property
+
+> **Issue**: [KT-48826](https://youtrack.jetbrains.com/issue/KT-48826)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: deprecate the `kapt.use.worker.api` property that allowed to run kapt via Kotlin daemon
```
Comment (Tapchicoma) [inline docs/topics/compatibility-guide-16.md:464]:
```suggestion
> **Short summary**: deprecate the `kapt.use.worker.api` property that allowed to run kapt via Gradle Workers API (default: true)
```

File: docs/topics/compatibility-guide-16.md [hunk 2]
```diff
@@ -452,3 +452,48 @@ perspective
...
+>
+> **Deprecation cycle**:
+>
+> - 1.6.20: raise the deprecation level to a warning
+> - \>= 1.8.0: raise this warning to an error
```
Comment (Tapchicoma) [inline docs/topics/compatibility-guide-16.md:469]:
```suggestion
> - \>= 1.8.0: remove this property
```

File: docs/topics/compatibility-guide-16.md [hunk 3]
```diff
@@ -452,3 +452,48 @@ perspective
...
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: remove the `kotlin.parallel.tasks.in.project` property
+>
+> **Deprecation cycle**:
+>
+> - 1.5.20: raise the deprecation level to a warning
+> - 1.6.20: raise this warning to an error
```
Comment (Tapchicoma) [inline docs/topics/compatibility-guide-16.md:484]:
```suggestion
> - 1.6.20: remove this property
```

File: docs/topics/compatibility-guide-16.md [hunk 4]
```diff
@@ -452,3 +452,48 @@ perspective
...
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: deprecate the `kotlin.experimental.coroutines` Gradle DSL option and the `kotlin.coroutines` property
+>
+> **Deprecation cycle**:
+>
+> - 1.6.20: raise the deprecation level to a warning
+> - \>= 1.7.0: raise this warning to an error
```
Comment (Tapchicoma) [inline docs/topics/compatibility-guide-16.md:499]:
```suggestion
> - \>= 1.7.0: remove DSL option and property
```

File: docs/topics/compatibility-guide-16.md [hunk 5]
```diff
@@ -452,3 +452,48 @@ perspective
...
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: deprecate the `kotlin.experimental.coroutines` Gradle DSL option and the `kotlin.coroutines` property
+>
+> **Deprecation cycle**:
+>
+> - 1.6.20: raise the deprecation level to a warning
+> - \>= 1.7.0: remove DSL option and property
```
Comment (PetrakovichVictoria) [inline docs/topics/compatibility-guide-16.md:499]:
```suggestion
> - \>= 1.7.0: remove the DSL option and the property
```

File: docs/topics/kapt.md [hunk 6]
```diff
@@ -106,20 +106,38 @@ kapt {
...
+To improve the speed of builds that use kapt, you can enable the [Gradle Worker API](https://guides.gradle.org/using-the-worker-api/)
+for kapt tasks. Using the Worker API lets Gradle run independent annotation processing tasks from a single project in parallel,
+which in some cases significantly decreases the execution time.
 
-To use the Gradle worker API for parallel execution of kapt tasks, add this line to your `gradle.properties` file:
+When you use the [custom JDK home](gradle.md#set-custom-jdk-home) feature in the Kotlin Gradle plugin,
+kapt task workers use only [process isolation mode](https://docs.gradle.org/current/userguide/worker_api.html#changing_the_isolation_mode).
+Note that the `kapt.workers.isolation` property is ignored.
+
+If you want to provide additional JVM arguments for a kapt worker process, use the input `kaptProcessJvmArgs` of the `KaptWithoutKotlincTask`:
```
Comment (PetrakovichVictoria) [inline docs/topics/kapt.md:117]: @Tapchicoma, I added about `kaptProcessJvmArgs` of the `KaptWithoutKotlincTask`

**Review by Tapchicoma — APPROVED** [review]

File: docs/topics/kapt.md (hunk 6 above)
Comment (PetrakovichVictoria) [inline docs/topics/kapt.md:117]: The new text starts from here.

**Review by p7nov — APPROVED** [review]
👍

**Pull request merged**